.buildkite/engineer

```diff
@@ -1,5 +1,32 @@
 #!/usr/bin/env bash
 
+set -e
+
+# Checks what's the diff with the previous commit, 
+# excluding some paths that do not need a run, 
+# because they do not affect tests running in Buildkite.
+GIT_DIFF=$(git diff --name-only HEAD HEAD~1 -- . ':!.github' ':!query-engine/driver-adapters/js' ':!renovate.json' ':!*.md' ':!LICENSE' ':!CODEOWNERS';)
+
+# $2 is either "test" or "build", depending on the pipeline
+# Example: ./.buildkite/engineer pipeline test
+# We only want to check for changes and skip in the test pipeline.
+if [[ "$2" == "test" ]]; then
+    echo "We are in the test pipeline."
+    # Checking if GIT_DIFF is empty
+    # If it's empty then it's most likely that there are changes but they are in ignored paths.
+    # So we do not start Buildkite
+    if [ -z "${GIT_DIFF}" ]; then
+        echo "No changes found for the previous commit in paths that are not ignored, this run will now be skipped."
+        exit 0
+    else
+        # Note that printf works better for displaying line returns in CI
+        printf "Changes found for the previous commit in paths that are not ignored: \n\n${GIT_DIFF}\n\nThis run will continue...\n"
+    fi
+else
+    echo "We are in the build pipeline."
```

Comment:
Is there a real intention to print this message when building? Or is it a leftover?

Reply:
It's to be clear/explicit about what's happening.
I made it more generic now